WrightTools/data/_data.py

```diff
@@ -427,16 +437,23 @@ def convert(self, destination_units, *, convert_variables=False, verbose=True):
         # apply to all compatible axes
         for axis in self.axes:
             if axis.units_kind == units_kind:
-                axis.convert(destination_units, convert_variables=convert_variables)
                 if verbose:
-                    print("axis", axis.expression, "converted")
+                    print(
+                        "axis {} converted from {} to {}".format(
+                            axis.expression, axis.units, destination_units
+                        )
+                    )
+                axis.convert(destination_units, convert_variables=convert_variables)
```

Comment:
@parkertomf can you move the call to `axis.convert` _before_ the `if verbose` switch?

that way the print statement is only made if the axis was actually converted successfully

Reply:
(note this applies to both lines here)